lib/dialects/abstract/query-generator.js

```diff
@@ -1211,22 +1214,14 @@ var QueryGenerator = {
                 if (topInclude.through && Object(topInclude.through.model) === topInclude.through.model) {
                   $query = self.selectQuery(topInclude.through.model.getTableName(), {
                     attributes: [topInclude.through.model.primaryKeyAttributes[0]],
-                    include: [{
-                      model: topInclude.model,
-                      as: topInclude.model.name,
-                      attributes: [],
-                      association: {
-                        associationType: 'BelongsTo',
-                        isSingleAssociation: true,
-                        source: topInclude.association.target,
-                        target: topInclude.association.source,
-                        identifier: topInclude.association.foreignIdentifier,
-                        identifierField: topInclude.association.foreignIdentifierField
-                      },
-                      required: true,
-                      include: topInclude.include,
-                      _pseudo: true
-                    }],
+                    include: Model.$validateIncludedElements({
```

Comment:
Not really, it's the validation that's normally called as part of a find call. We have to call it here since we call selectQuery directly, the validation does all sorts of stuff like figure out if we need a subquery, what associations should be in the subquery etc.